presto-hive/src/test/java/com/facebook/presto/hive/TestHiveLogicalPlanner.java

```diff
@@ -2569,6 +2571,89 @@ public void testMaterializedViewForJoinWithMultiplePartitions()
         }
     }
 
+    @Test
+    public void testMaterializedViewResultForTwoTableJoin()
+    {
+        final String viewName = "mv";
+        final String table1 = "lineitem_partitioned";
+        final String table2 = "orders_partitioned";
+
+        @Language("SQL") final String joinSelect = format("SELECT custkey, %s.orderkey, orderstatus, linenumber " +
+                "FROM %s JOIN %s " +
+                "ON %s.orderkey = %s.orderkey " +
+                "ORDER BY custkey, %s.orderkey, orderstatus, linenumber ",
+                table1, table1, table2, table1, table2, table1);
+        QueryRunner queryRunner = getQueryRunner();
+        try {
+            queryRunner.execute(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['linenumber']) AS " +
+                    "SELECT orderkey, linenumber FROM lineitem", table1));
+            queryRunner.execute(format("CREATE TABLE %s WITH (partitioned_by = ARRAY['orderstatus']) AS " +
+                    "SELECT orderkey, custkey, orderstatus FROM orders", table2));
+            assertUpdate(format(
+                    "CREATE MATERIALIZED VIEW %s " +
+                            "WITH (partitioned_by = ARRAY['orderstatus', 'linenumber'])" +
+                            "AS %s",
+                    viewName,
+                    joinSelect));
+
```

Comment:
Is it possible to fire some refresh commands to insert some values in the materialized view?